src/gengowatcher/event_bus.py

```
@@ -0,0 +1,103 @@
+    with _CONSUMER_LOCK:
+        consumers = list(_CONSUMERS.items())
+
+    event_dict = event.to_dict()
+
+    for name, q in consumers:
+        if coalesce and event.type == "job.status" and event.collection_id:
+            # Use bus-owned state to check for recent duplicate
+            # instead of peeking at queue internals
+            should_skip = False
+            with _CONSUMER_LOCK:
+                last_seen = _coalesce_last_seen.get(name)
```

Ruff (0.15.20)

[error] Undefined name `_coalesce_last_seen`

(F821)